alg_hmac/key_dir.go

```diff
@@ -0,0 +1,50 @@
+package alg_hmac
+
+import (
+	"context"
+	"errors"
+
+	"github.com/common-fate/httpsig/verifier"
+)
+
+type singleHMACKeyDirectory struct {
+	hmac *HMAC
+}
+
+// NewSingleHMACKeyDirectory creates a new in-memory SingleHMACKeyDirectory with the provided key.
+// All parameters are ignored in the GetKey() method, and it will never error.
+func NewSingleHMACKeyDirectory(hmac *HMAC) verifier.KeyDirectory {
```

Comment:
`SingleHMACKeyDirectory` deviates from the naming used in the `alg_ecdsa` package (`StaticKeyDirectory`). "Single" and "Multi" are more concise names though, I prefer the naming introduced here. I will go back and update the name in `alg_ecdsa` to match these.